### PR TITLE
Ee 16843 get audit summary endpoint

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
Added an endpoint that will return all the Archived Audit Summaries between two dates. As this has added a new endpoint onto audit-service I would like it to be tested before it is merged to master. The JIRA for testing is EE-17337

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/pttg/api/ArchiveResource.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        this.archiveService = archiveService;
+    }
+
+    @GetMapping(value = "/archived", produces = APPLICATION_JSON_VALUE)
+    public List<ArchivedResult> getArchivedResults(
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate fromDate,
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate toDate
+    ) {
+        log.info("Request received for archived results between {} and {}", fromDate, toDate,
+                value(EVENT, PTTG_AUDIT_ARCHIVE_REQUEST_RECEIVED));
```
Comment (mikehalmaho) [inline src/main/java/uk/gov/digital/ho/pttg/api/ArchiveResource.java]: The event sounds like it is a request to create an archive... maybe just changing ARCHIVE_REQUEST to GET_ARCHIVED_REQUEST or similar?

File: src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @Test
+    public void shouldReturnArchivedResults() throws JsonProcessingException {
+        List<ArchivedResult> expected = asList(
+                new ArchivedResult(ImmutableMap.of("PASS", 5, "FAIL", 3)),
+                new ArchivedResult(ImmutableMap.of("ERROR", 2, "NOTFOUND", 99))
+        );
+
+        String archivedResults = restTemplate.getForObject("/archived?fromDate={fromDate}&toDate={toDate}", String.class, ImmutableMap.of(
+                "fromDate", LocalDate.of(2018, Month.SEPTEMBER, 1),
+                "toDate", LocalDate.of(2018, Month.SEPTEMBER, 30)
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java]: Would there be a benefit in making at least one or both of the dates on the boundary of the test data?

File: src/main/java/uk/gov/digital/ho/pttg/api/ArchiveResource.java (hunk 1 above)
Comment (owainrobertsbjss) [reply src/main/java/uk/gov/digital/ho/pttg/api/ArchiveResource.java]: Agreed

File: src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java (hunk 2 above)
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java]: Yes - I have adjusted the data to improve the test

File: src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceTest.java [hunk 3]
```diff
@@ -77,10 +86,69 @@ public void archiveResult_logsSuccess() {
...
+                        SOME_ARCHIVED_RESULT,
+                        SOME_ARCHIVED_RESULT
+                ));
+
+        archiveResource.getArchivedResults(SOME_DATE, SOME_DATE);
+
+        assertInfoLog("Returned response with 3 archived results", PTTG_AUDIT_GET_ARCHIVED_RESULTS_RESPONSE_SUCCESS);
+    }
+
+    public void assertInfoLog(String expectedMessage, LogEvent logEvent) {
```
Comment (Tim-Hammonds) [inline src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceTest.java]: Can the "archiveResult_logs*" tests use this helper method?

File: src/main/java/uk/gov/digital/ho/pttg/api/ArchiveResource.java [hunk 4]
```diff
@@ -24,6 +26,22 @@ public ArchiveResource(ArchiveService archiveService, RequestData requestData) {
         this.requestData = requestData;
     }
 
+    @GetMapping(value = "/archive", produces = APPLICATION_JSON_VALUE)
+    public List<ArchivedResult> getArchivedResults(
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate fromDate,
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate toDate
+    ) {
```
[the remark this answers is not in the document]
Comment (owainrobertsbjss) [reply src/main/java/uk/gov/digital/ho/pttg/api/ArchiveResource.java]: IDE seems happy. Leaving as discussed

File: src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+@SqlGroup({
+        @Sql(executionPhase = BEFORE_TEST_METHOD, scripts = "/sql/ArchiveResourceIntTest/before.sql"),
+        @Sql(executionPhase = AFTER_TEST_METHOD, scripts = "/sql/ArchiveResourceIntTest/after.sql")
+})
+public class ArchiveResourceIntTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
```
[the remark this answers is not in the document]
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java]: Agreed and addressed

File: src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceTest.java (hunk 3 above)
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceTest.java]: I've done archiveResult_logsRequestParameters but not archiveResult_logsSuccess as the latter checks for request_duration_ms in the argument array too.

**Pull request merged**